testsuite/model/src/test/java/org/keycloak/testsuite/model/transaction/StorageTransactionTest.java

```diff
@@ -200,8 +200,9 @@ private LockObjectsForModification.CallableWithoutThrowingAnException<UserSessio
     }
 
     @Test
-    // Optimistic locking works only with map-jpa
-    @RequireProvider(value = MapStorageProvider.class, only = JpaMapStorageProviderFactory.PROVIDER_ID)
+    // Optimistic locking works only with map-jpa and map-hotrod
```

Comment:
As reading by id and reading by query differs, should we have two variants (reading by id and by query) for each optimistic locking test?

Reply:
Replace and delete are two different operations. I would say it makes sense to include both as well as test for removing when reading by `id`. This could make sense also for other storages. Maybe some storage won't have optimistic locking for delete but only for replace.